doc/style-guide.md

```diff
@@ -80,6 +140,21 @@ We use UTF-8 everywhere, including on Windows where UTF-16 is the more common ap
 * On Windows, when using Win32 and similar APIs, we must convert UTF-8 strings to UTF-16 and then call the wide character version of the system API (e.g. CreateFileW). Do this at the call site.
 * Be careful when using the `fstream` API family on Windows. Make sure you read and understand [How to do text on Windows](https://utf8everywhere.org/#windows).
 
+## ✅ Testing
+
+Not covered by C++ Core Guidelines.
+
+* Test files should be prefixed with `Test`, e.g. `TestModel` instead of `ModelTests`.
+
+## 🗂️ Other
+
+Not covered by C++ Core Guidelines.
+
+* Use `this->` when accessing member variables and functions. This helps distinguish member variables from local variables and member functions from anonymous namespace functions.
+* Static function definitions in a .cpp file should be preceded by `/*static*/`.
+* Unused parameter names in callbacks should be commented out rather than omitted. Only use `[[maybe_unused]]` when you can't comment out parameter names, such as when parameters are conditionally accessed by `constexpr` logic.
+* Use `[i]` instead of `.at(i)` for sequence containers like `std::vector` and `std::array`.
```

Comment:
I considered splitting these into separate sections, but it felt like too many extra headings. What do others think?